### PR TITLE
Setup ingress out-of-box using the cluster install config

### PR DESCRIPTION
**Pull request by ramr**
Ref: https://jira.coreos.com/browse/NE-88

Ran into a few issues whilst testing this (hence the other commits) - I can break that out separately or squash if needed. 
Am not certain on the finalizer permission set  (last commit/sha ad93bca). Seems like there was code added for setting an owner object on the actual router service but not really tested. 

@ironcladlou @openshift/sig-network-edge  PTAL  Thx

### CODE REVIEW

File: pkg/util/config.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+	if !ok {
+		return nil, fmt.Errorf("invalid installer mao-config provider")
+	}
+	haType := ingressv1alpha1.ClusterIngressHAType("")
+	if maoConfigProvider == "aws" {
+		haType = ingressv1alpha1.CloudClusterIngressHA
+	}
+
+	return &ConfigDefaults{
+		IngressDomain: fmt.Sprintf("apps.%s.%s", metadataName, baseDomain),
```
[the remark this answers is not in the document]
Comment (ramr) [reply pkg/util/config.go]: yeah I saw that - will get it from there.

File: manifests/02-deployment.yaml [hunk 2]
```diff
@@ -16,6 +16,7 @@ spec:
       containers:
         - name: cluster-ingress-operator
           image: openshift/origin-cluster-ingress-operator:latest
+          imagePullPolicy: IfNotPresent
```
[the remark this answers is not in the document]
Comment (ramr) [reply manifests/02-deployment.yaml]: So it was always pulling the image (and I couldn't get it to use a locally built image).

**Comment by ramr** [comment]
@ironcladlou PTAL reworked as per your suggestions. Tested this out locally  + added some unit tests. Am out tomorrow but can follow up on this if needed. Thx

**Comment by ironcladlou** [comment]
Looking good; I'll try testing it against a cluster myself.

**Comment by ironcladlou** [comment]
Tested this end-to-end with a new 4.0 cluster:

1. Created new cluster
2. [Uninstalled the CVO and existing ingress setup](https://gist.github.com/ironcladlou/d3a7af553130f710a08475adc4096624#file-uninstall-sh)
3. Installed the cluster-ingress-operator from manifests
4. Created an app which publishes a route admitted by the new default router
5. Tested connectivity to the app over the router's public DNS name (using `Host` header for the routing domain)

Awesome, thanks!

/lgtm

**Pull request merged**